qadence/constructors/rydberg_feature_maps.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+from sympy import Basic, Function
+
+from qadence.blocks import AnalogBlock, KronBlock, kron
+from qadence.constructors.feature_maps import fm_parameter
+from qadence.logger import get_logger
+from qadence.operations import AnalogRot, AnalogRX, AnalogRY, AnalogRZ
+from qadence.parameters import FeatureParameter, Parameter, VariationalParameter
+from qadence.types import BasisSet, TParameter
+
+logger = get_logger(__file__)
+
+AnalogRotationTypes = [AnalogRX, AnalogRY, AnalogRZ]
+
+
+def rydberg_feature_map(
+    n_qubits: int,
+    param: str = "phi",
+    max_abs_detuning: float = 2 * np.pi * 10,
+    weights: list[float] | None = None,
+) -> KronBlock:
+    """Feature map using semi-local addressing patterns.
+
+    If not weights are specified, variational parameters are created
+    for the pattern
+
+    Args:
+        n_qubits (int): number of qubits
+        param: the name of the feature parameter
+        max_abs_detuning: maximum value of absolute detuning for each qubit
+        weights: a list of wegiths to assign to each qubit parameter in the feature map
+
+    Returns:
+        The block representing the feature map
+    """
+    max_abs_detuning = 2 * np.pi * 10
+
+    tower_coeffs: list[float | Parameter]
+    tower_coeffs = (
+        [VariationalParameter(f"w_{param}_{i}") for i in range(n_qubits)]
+        if weights is None
+        else weights
+    )
+    tower_detuning = max_abs_detuning / (sum(tower_coeffs[i] for i in range(n_qubits)))
+
+    param = FeatureParameter(param)
+    duration = 1000 * param / tower_detuning
+    return kron(
+        AnalogRot(
+            duration=duration,
+            delta=-tower_detuning * tower_coeffs[i],
+            phase=0.0,
+            qubit_support=(i,),
+        )
+        for i in range(n_qubits)
+    )
+
+
+def rydberg_tower_feature_map(
+    n_qubits: int, param: str = "phi", max_abs_detuning: float = 2 * np.pi * 10
+) -> KronBlock:
+    weights = list(np.arange(1, n_qubits + 1))
+    return rydberg_feature_map(
+        n_qubits, param=param, max_abs_detuning=max_abs_detuning, weights=weights
+    )
+
+
+def analog_feature_map(
+    param: str = "phi",
+    op: Callable[[Parameter | Basic], AnalogBlock] = AnalogRX,
+    fm_type: BasisSet | type[Function] | str = BasisSet.FOURIER,
+    feature_range: tuple[float, float] | None = None,
+    target_range: tuple[float, float] | None = None,
+    multiplier: Parameter | TParameter | None = None,
+) -> AnalogBlock:
```

Comment:
One thing I did in the QCL example in the documentation that I seem to recall helped with training was to reupload data by applying a few different analog rotations in sequence with a different reupload scaling. This could be built into this function.

But anyway, this can be expanded on in a separate MR.